### PR TITLE
[FD-3394] Edit Profile

### PR DESCRIPTION
**Pull request by xixe**

### CODE REVIEW

File: src/modules/profile/ProfileRoutes.tsx [hunk 1]
```diff
@@ -39,6 +51,12 @@ export function ProfileRoutes() {
         exact={true}
         component={LoadableProfileContainer}
       />
+
+      <PrivateRoute
```
Comment (xixe) [inline src/modules/profile/ProfileRoutes.tsx:55]: Something wrong with this route. When route is '/profile/edit' component for '/profile' is visible too.

Comment (xixe) [reply src/modules/profile/ProfileRoutes.tsx]: I think, I resolve this with wrap these routes in <Switch>. However it was strange

**Review by Antroll — APPROVED** [review]

File: src/modules/form/components/InputField/InputField.tsx [hunk 2]
```diff
@@ -1,24 +1,41 @@
...
-import { FieldRenderProps } from 'react-final-form';
+import React, { useCallback } from 'react';
+import { FieldMetaState, FieldRenderProps } from 'react-final-form';
 import { getErrorText } from '../../utils/getErrorText';
 import { hasError } from '../../utils/hasError';
+import { useInputFieldStyles } from './InputFieldStyles';
 
-interface IFieldProps extends FieldRenderProps<string> {}
+interface IFieldProps extends FieldRenderProps<string> {
+  showlimitcounter: boolean;
```
Comment (BigBadAlien) [inline src/modules/form/components/InputField/InputField.tsx:9]: showLimitCounter

File: src/modules/form/components/InputField/InputField.tsx [hunk 3]
```diff
@@ -1,24 +1,41 @@
...
   ...rest
 }: IFieldProps & TextFieldProps) => {
+  const classes = useInputFieldStyles();
+
+  const maxLength:number|null = rest.inputProps?.maxLength ?? null;
+
+  const getHelperString = useCallback((value:string, meta:FieldMetaState<any>):string => {
+    let helperTextString:string = getErrorText(meta);
+    if (showlimitcounter && maxLength && !hasError(meta)) {
+      helperTextString = `${value.length ?? 0}/${maxLength}`;
```
Comment (BigBadAlien) [inline src/modules/form/components/InputField/InputField.tsx:25]: t

File: src/modules/profile/screens/EditProfile/EditProfile.tsx [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+    errors.username = t('validation.required');
+  }
+
+  return errors;
+};
+
+const handleSubmit = () => {};
+
+export const EditProfile = () => {
+  const BIO_CHARACTER_LIMIT = 200;
```
Comment (BigBadAlien) [inline src/modules/profile/screens/EditProfile/EditProfile.tsx:35]: Move out plz

File: src/modules/profile/screens/EditProfile/EditProfile.tsx [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+            type='text'
+            label={t('profile.edit.label.username')}
+            color='primary'
+            fullWidth={true}
+          />
+        </Box>
+        <Box mb={5}>
+          <Field
+            component={InputField}
+            name='email'
```
Comment (BigBadAlien) [inline src/modules/profile/screens/EditProfile/EditProfile.tsx:53]: validation

File: src/modules/profile/screens/EditProfile/EditProfile.tsx [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+            placeholder={t('profile.edit.placeholder.facebook')}
+            color='primary'
+            fullWidth={true}
+          />
+        </Box>
+
+        <Box>
+          <Mutation>
+            {({ loading }) => (
+              <Button size='large' type='submit' fullWidth disabled={loading}>
```
Comment (BigBadAlien) [inline src/modules/profile/screens/EditProfile/EditProfile.tsx:133]: Create action, dispatch it and pass to Mutation as type

**Review by BigBadAlien — APPROVED** [review]

File: src/modules/profile/screens/EditProfile/EditProfile.tsx [hunk 7]
```diff
@@ -0,0 +1,191 @@
...
+}
+
+const validateEditProfile = (payload: IEditProfile) => {
+  const errors: FormErrors<IEditProfile> = {};
+
+  if (!payload.username) {
+    errors.username = t('validation.required');
+  }
+
+  const emailRegex = /^(([^<>()\\.,;:\s@"]+(\.[^<>()\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
```
Comment (BigBadAlien) [inline src/modules/profile/screens/EditProfile/EditProfile.tsx:32]: Extract into function plz

e.g. /common/utils/isValidaEmail.ts

File: src/modules/profile/screens/EditProfile/EditProfile.tsx [hunk 8]
```diff
@@ -0,0 +1,191 @@
...
+  if (!!payload.email && !emailRegex.test(String(payload.email).toLowerCase())) {
+    errors.email = t('validation.invalid-email');
+  }
+
+  const urlRegex = /^(?:(?:http|https):\/\/)(?:\\S+(?::\\S*)?@)?(?:(?:(?:[1-9]\\d?|1\\d\\d|2[01]\\d|22[0-3])(?:\\.(?:1?\\d{1,2}|2[0-4]\\d|25[0-5])){2}(?:\\.(?:[0-9]\\d?|1\\d\\d|2[0-4]\\d|25[0-4]))|(?:(?:[a-z\\u00a1-\\uffff0-9]+-?)*[a-z\\u00a1-\\uffff0-9]+)(?:\\.(?:[a-z\\u00a1-\\uffff0-9]+-?)*[a-z\\u00a1-\\uffff0-9]+)*(?:\\.(?:[a-z\\u00a1-\\uffff]{2,})))|localhost)(?::\\d{2,5})?(?:(\/|\\?|#)[^\\s]*)?$/;
```
Comment (BigBadAlien) [inline src/modules/profile/screens/EditProfile/EditProfile.tsx:37]: The same, extract plz

**Pull request merged**